plutus-core/plutus-core/src/Crypto.hs

```diff
@@ -17,3 +34,64 @@ verifySignature pubKey msg sig =
             <$> publicKey pubKey
             <*> pure msg
             <*> signature sig
+
+-- | Verify an ECDSA signature made using the SECP256k1 curve.
+--
+-- = Note
+--
+-- This takes a message /hash/, rather than a general blob of bytes; thus, it is
+-- limited in length.
+verifyEcdsaSecp256k1Signature
+  :: BS.ByteString -- ^ Public key (64 bytes)
```

Comment:
I've added [similar comments](https://github.com/input-output-hk/plutus/pull/4437) about input lengths to the existing `verifySignature` function, and we'll update it to use the `DSIGN` interface at some point as well.  We should probably also give it a more informative name (or at least an alternative) if we can do that without breaking anything.

Reply:
I believe @michaelpj said that it'd be too breaking to change the name somewhere else.